### PR TITLE
[WIP] 734 layout changes on resource specific tab

### PR DESCRIPTION
**Pull request by mauriel133**
These are layout changes for the resource specific tab for each resource type. This changes make efficient use of the space provided and accommodates for the removal of the max height that restrained the tab. Also includes minor style changes for consistency.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by rayi113** [comment]
@pkdash to list who can review which resource templates

**Comment by pkdash** [comment]
@mohamedmorsy review changes to model instance template changes

**Comment by pkdash** [comment]
@Castronova please review template/js code changes for model program

**Comment by pkdash** [comment]
@gantian127 please review code changes for netcdf and raster resource types

File: hs_modelinstance/templates/pages/modelinstanceresource.html [hunk 1]
```diff
@@ -6,71 +6,52 @@
...
-                    </div>
-
+                    <div class="col-sm-8">No</div>
                 {% endif %}
 
-
             </div>
-        </div>
-        </div>
+        {% endif %}
```
Comment (mohamedmorsy) [inline hs_modelinstance/templates/pages/modelinstanceresource.html]: Could we add "<br>" after "{% endif %}" to do a separation between ModelOutput and ExecutedBy?

File: hs_modelinstance/templates/pages/modelinstanceresource.html [hunk 2]
```diff
@@ -6,71 +6,52 @@
...
+            </div>
+        {% else %}
+          <legend style="margin-top:40px;">Executed By</legend>
+            <div class="row">
+                    <div class="col-sm-4"><strong>Name</strong></div>
+                    <div class="col-sm-8">Unspecified</div>
+            </div>
+
+        {% endif %}
     {% endif %}
```
Comment (mohamedmorsy) [inline hs_modelinstance/templates/pages/modelinstanceresource.html]: If you do the suggestion above, could you also add `<br>` after `{% endif %}` to make the line spacing concise in the page?

**Comment by mohamedmorsy** [comment]
@mauriel133 Could you please merge the new modifications in the develop branch? As I would like to see how these new modifications in this branch are going to affect the new implementation of the Model Instance and SWAT Model Instance resources.

**Comment by gantian127** [comment]
@mauriel133 I have made a comment for issue

**Comment by hyi** [comment]
@mauriel133 What is the current status for this PR?

**Comment by mauriel133** [comment]
I'm applying some changes after feedback from other programmers.

**Comment by hyi** [comment]
@mauriel133 OK. I am planning to release 1.6.3 this week if possible, so please let me know when it is ready to be merged and whether anyone else needs to review the layout change before merging.

**Comment by mauriel133** [comment]
I have finished applying changes now. I think all resources look fine under resource specific tab of the landing page. 

@mohamedmorsy, I merged the changes in the develop branch and applied the new style, so those can be tested now.

**Comment by hyi** [comment]
@mauriel133 @gantian127 @mohamedmorsy @zhiyuli I deployed this to park.hydroshare.org and created one raster resource and tested layout spacing for resource specific tab and it does not look correct to me. Please test them out for your own resource types and let @mauriel133 know if anything does not look correct. @mauriel133 after you apply further changes, let me know and I can apply new changes to park for continuous testing by others until they give thumbs up. Then I can merge this to develop branch.

**Comment by gantian127** [comment]
@mauriel133 @hyi netCDF resource layout is working as expected. Thanks.

**Comment by hyi** [comment]
@gantian127 How about raster resource layout?

**Comment by mauriel133** [comment]
On raster resources, the "Coordinate Refetence System" label is repeated.

**Comment by gantian127** [comment]
@mauriel133 @hyi 
Do you mean these two labels are repeated? I think the second means the "coordinate reference system unit" for which it may put the "unit" on the same line
qq 20151207103842

Moreover,  I found out that sometimes under editing mode, the fields which need to be editable for the spatial reference and cell information are not able to be edited.

**Comment by gantian127** [comment]
@mauriel133 I made a public raster resource with multiple bands in the website and you can test with that file. The fields for the spatial reference and cell information are not editable which is not working correctly.

Another suggestion is for Band information. It may be better to list them starting with "band_1". If this suggestion is complicated to implement, you can just leave it as it is. Thanks

**Comment by mauriel133** [comment]
Noticed. I will look into these issues.

**Comment by mauriel133** [comment]
@gantian127, I cannot reproduce the layout error with multiple bands. Can you make sure that it wasn't due to outdated cached files? 

If it keeps happening tell me which browser and version you are using.

**Comment by gantian127** [comment]
@mauriel133 I think the layout under view mode is working correctly now for band information. Maybe it just happened on my computer yesterday. Thanks

**Comment by hyi** [comment]
@mauriel133 @gantian127 @zhiyuli @mohamedmorsy I have redeployed the latest change of this branch to park.hydroshare.org, so you can take a look as needed. @mauriel133 You have to rebase this branch with latest develop to resolve conflicts since it cannot be merged automatically now. Let me know when it is ready to merge.

**Comment by mauriel133** [comment]
Done.

**Comment by gantian127** [comment]
@mauriel133 I found out that the "variable" forms of netcdf resource under editing mode is missing. Please check this see if you find the same error. Thanks.

**Comment by gantian127** [comment]
@mauriel133 for raster resource. Under editing mode, the band information "save changes" button doesn't take action and the edited information can't be saved. 
Under view mode, is it possible to make the " coordinate reference system unit" on the same line, so that user won't get confused to think that the terms are repeated. Thanks.

**Comment by mohamedmorsy** [comment]
@mauriel133  <https://github.com/mauriel133> I found that in the Model
Instance specific metadata under "Executedby" term, the version of the
Model Program used to execute this instance is available in the edit mode
while it is not available in the resource landing page:

[image: Inline image 1]

[image: Inline image 2]

On Thu, Dec 10, 2015 at 3:01 PM, gantian127 <notifications@github.com>
wrote:

> @mauriel133 <https://github.com/mauriel133> I found out that the
> "variable" forms of netcdf resource under editing mode is missing. Please
> check this see if you find the same error. Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hydroshare/hydroshare/pull/740#issuecomment-163734515>
> .
>



-- 
Mohamed M. Morsy
Graduate Research Assistant
Department of Civil & Environmental Engineering
Thornton Hall D107, University of Virginia
351 McCormick Road, P.O. Box 400742
Charlottesville, VA 22904-4742
Phone: (434) 924-7630
E-mail: mmm4dh@virginia.edu

**Comment by mohamedmorsy** [comment]
@mauriel133 For the SWAT Model Instance, it has the same old style of the specific metadata landing page (the style did not updated to be look like the one for the Model Instance resource). Also it has the same problem I stated before for the Model Program Version used for execution ("Not Provided"):

image

In the editing mode for the specific metadata landing page, there is a wide white gap between "Model Program used for execution" and "Model Method" terms:

image

There is a big wide white gap between "Model Method" and "Model Parameter" terms:

image

**Comment by mauriel133** [comment]
I cannot set a specific width for " coordinate reference system unit" because it would conflict with the responsive layout for mobile devices.

@mohamedmorsy, I'll look into those issues. Thanks.

**Comment by gantian127** [comment]
@mauriel133 that will fine if the width can't be set. I will test the update after it is deployed. Thanks for working on these issues.

**Comment by hyi** [comment]
@mauriel133 @gantian127 @zhiyuli @mohamedmorsy The latest code in the branch has just been deployed on park for your further testing. Please raise issues or give +1 for your resource types so that we know when it is ready to merge this.

**Comment by gantian127** [comment]
+1 for netcdf and raster resource

**Comment by zhiyuli** [comment]
@mauriel133  @hyi  +1 for Geog. Feature Res. Type

**Comment by hyi** [comment]
@mohamedmorsy @mauriel133 could you check your model-related resource types on park and see whether things look good now? Will merge this in for AGU if you give a +1.

**Comment by mohamedmorsy** [comment]
@hyi I have two issues related to Model Instance and SWAT Model Instance resources and I am waiting for @mauriel133 to take a look on that.

**Comment by hyi** [comment]
@mohamedmorsy OK. @mauriel133 will not have time until he finishes getting my resources page ready for AGU. @mauriel133 you don't have to get this in for pre-deployment on beta today, and since this branch only involves front-end style changes, this can be easily pulled into AGU release on Monday before final www deployment.

**Comment by mauriel133** [comment]
@mohamedmorsy 
I have applied the style to SWAT Model Instance resources. Please look at it and let me know if you find any issues.

The issue with the Version field seems to be caused by something else. I'm using the exact same conditional checks to render it, and the issue is also present in the current develop branch as deployed on beta prior to this work.

This is how it is currently implemented:
>> {% if executed_by.model_program_fk.metadata.program.software_version.strip %}
                                    {{ executed_by.model_program_fk.metadata.program.software_version }}
                                {% else %}
                                    Not Provided
                                {% endif %}
Perhaps adding ".strip" at the end of the line would make it show up. Could you try that and see if it works, otherwise another issue should be opened for this so that this PR can move on with the rest of the work.

**Comment by mohamedmorsy** [comment]
@mauriel133 @hyi  I pulled the branch on my local computer and tested it. I found the bug related to the version field. We got this bug because we merged the new version of the model instances before the new version of the model program. So in the new version of the model program there is no field called software_version as it becomes "modelVersion". I did the fix and pushed it to this branch as it is a minor fix. Also I changed the name of the "Resource URL" field under the "ExecutedBy" to "Resource URI"
@mauriel133 I tested everything for the three model resorces, everything looks great and nice except for the SWAT Model Instance landing page. As you see in the following screenshot, the "Resource URI" field under "ExecutedBy" is interfering with the "ModelInput" fields. Also it would be great if we can put "Resource URI" in the same line. Also it would be great if we can put the name of the fields of the "ModelInput" metadata term on the same line.

image

**Comment by mauriel133** [comment]
@mohamedmorsy, I will move the Model Input section to the bottom and have everything in one column in order to avoid shortage of space in the field names. 

Thanks for the feedback and for taking care of the bug.

**Comment by mauriel133** [comment]
You can test the changes now.

**Comment by mohamedmorsy** [comment]
@mauriel133 Now, everything looks great! Thanks for your hard work on that!

@hyi +1 for Model resources

**Pull request merged**